### PR TITLE
feat: Add Onedrive Trigger Node

### PR DESCRIPTION
**Pull request by bramkn**
## Summary
Adds the Onedrive Trigger Node

## Related tickets and issues
Linear: https://linear.app/n8n/issue/NODE-1158/onedrive-trigger-node-new-node

## Review / Merge checklist
- [ ] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

File: packages/nodes-base/nodes/Microsoft/OneDrive/MicrosoftOneDriveTrigger.node.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
+import type {
```
Comment (maspio) [inline packages/nodes-base/nodes/Microsoft/OneDrive/MicrosoftOneDriveTrigger.node.ts:1]: Missing a MicrosoftOneDriveTrigger.node.json codex metadata file.

File: packages/nodes-base/nodes/Microsoft/OneDrive/MicrosoftOneDriveTrigger.node.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+export class MicrosoftOneDriveTrigger implements INodeType {
+	description: INodeTypeDescription = {
+		displayName: 'Microsoft OneDrive Trigger',
+		name: 'microsoftOneDriveTrigger',
+		icon: 'file:oneDrive.svg',
+		group: ['trigger'],
+		version: 1,
+		description: 'Trigger for Microsoft OneDrive API.',
+		subtitle: '={{"Microsoft OneDrive Trigger"}}',
```
Comment (maspio) [inline packages/nodes-base/nodes/Microsoft/OneDrive/MicrosoftOneDriveTrigger.node.ts:21]: Subtitle would be better using `{{($parameter["event"])}}`

File: packages/nodes-base/nodes/Microsoft/OneDrive/TriggerDescription.ts [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+	type: 'resourceLocator',
+	default: { mode: 'id', value: '', cachedResultName: '' },
+	required: true,
+	modes: [
+		{
+			displayName: 'Link',
+			name: 'url',
+			type: 'string',
+			placeholder:
+				'e.g. https://onedrive.live.com/?WT.mc_id=PROD%5FOL%2DWeb%5FInApp%5FLeftNav%5FFreeOfficeBarOD&ocid=PROD%5FOL%2DWeb%5FInApp%5FLeftNav%5FFreeOfficeBarOD&id=170B5C65E30736A3%21136&cid=170B5C65E30736A3',
```
Comment (maspio) [inline packages/nodes-base/nodes/Microsoft/OneDrive/TriggerDescription.ts:64]: For me simpler urls like `https://onedrive.live.com/?id=4A4A05756B39BA4A%2112047&cid=4A4A05756B39BA4A` seem are the default.

File: packages/nodes-base/nodes/Microsoft/OneDrive/constants.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export const MICROSOFT_DRIVE_FILE_URL_REGEX =
+	'https:\/\/onedrive.live.com(?:\/.*?(\&|\?)(id=|resid=))(.+)(?:\&.*)';
```
Comment (maspio) [inline packages/nodes-base/nodes/Microsoft/OneDrive/constants.ts:2]: This regex is invalid and doesn't work. When trying use `Watch A Selected File` with RLC URL the input field is also vanishing due to this.

**Review by maspio — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.34.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.34.0)

**Pull request merged**